eventnotifier/http/http.go

```diff
@@ -159,25 +167,58 @@ func (n *Notifier) Notify(dis v1beta1.Disruption, event corev1.Event, notifType
 		req.Header.Add(headerKey, headerValue)
 	}
 
-	res, err := n.client.Do(req)
+	if n.authTokenProvider != nil {
+		token, err := n.authTokenProvider.AuthToken(context.Background())
+		if err != nil {
+			return fmt.Errorf("http notifier: unable to retrieve auth token through helper: %w", err)
+		}
+
+		req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", token))
+	}
+
+	n.logger.Debugw("http notifier: sending notifier event to http", "disruption", dis.Name, "eventType", event.Type, "message", notif.EventMessage)
 
+	res, err := n.client.Do(req)
 	if err != nil {
 		return fmt.Errorf("http notifier: error when sending notification: %w", err)
 	}
 
-	n.logger.Debugw("notifier: sending notifier event to http", "disruption", dis.Name, "eventType", event.Type, "message", notif.EventMessage)
+	n.logger.Debugw("http notifier: received response from http", "status", res.StatusCode)
 
 	if res.StatusCode >= 300 || res.StatusCode < 200 {
 		return fmt.Errorf("http notifier: receiving %d status code from sent notification", res.StatusCode)
 	}
 
-	if err = res.Body.Close(); err != nil {
-		return fmt.Errorf("http notifier: error when sending notification: %w", err)
-	}
+	defer func() {
+		if err = res.Body.Close(); err != nil {
+			n.logger.Warnw("http notifier: error closing body", "error", err)
+		}
+	}()
 
 	return nil
 }
 
+// splitHeaders parses the headers from a slice of strings in the format key:value.
+// It returns a map of headers or an error if the headers are not in the correct format.
+func splitHeaders(headers []string) (map[string]string, error) {
+	parsedHeaders := make(map[string]string)
+
+	for _, header := range headers {
+		if header == "" {
+			continue
+		}
+
+		splittedHeader := strings.Split(header, ":")
+		if len(splittedHeader) == 2 {
+			parsedHeaders[splittedHeader[0]] = splittedHeader[1]
+		} else {
+			return nil, fmt.Errorf("invalid headers: Must be in the format: key:value, found %s", header)
+		}
```

Comment:
we probably can ignore this error. Not worth throwing everything else away because of a faulty pair.

Reply:
Incorrect headers could lead to failed authentication or improper request handling, which could compromise the system's functionality and security. Thus, I am not sure it is correct to just dismiss it.